iwyu_output.cc

```diff
@@ -605,6 +636,28 @@ void IwyuFileInfo::ReportForwardDeclareUse(SourceLocation use_loc,
   LogSymbolUse("Marked fwd-decl use of decl", symbol_uses_.back());
 }
 
+void IwyuFileInfo::ReportUsingDeclUse(SourceLocation use_loc,
+                                      const UsingDecl* using_decl,
+                                      const NamedDecl* target_decl,
+                                      bool in_cxx_method_body,
+                                      const char* comment) {  
+  for(auto&& saved_decl : using_decl_lines_) {
+    if(saved_decl.matches(using_decl)) {
+      saved_decl.set_referenced();
+      break;
+    }
+  }
+
+  ReportFullSymbolUse(use_loc, using_decl, in_cxx_method_body, comment);
```

Comment:
This is not for using decl's from headers.  It's for using decl's in source. What this is doing is saying 'I have just accessed something through a using decl', which means we don't need to include anything to make that using decl valid because the actual use of it will force the correct include.  Here's an example;

#include <vector>
#include <list>
using std::swap;
// EOF

in the above case, std::swap is never referenced, so we will need to include vector or list to keep the file compiling.  (the old behaviour was to include both vector and list)

If we change the code;

#include <vector>
#include <list>
using std::swap;

void foo()
{
   std::vector<int> a, b;
   swap(a, b); // This will call ReportUsingDeclUse
}

Now we know which file we need to include; vector.

Does this make sense?